src/core/meshcop/meshcop_tlvs.cpp

```diff
@@ -196,11 +182,6 @@ otError ChannelMaskTlv::SetChannelMask(uint32_t aChannelMask)
         entry->SetMask(aChannelMask & OT_RADIO_915MHZ_OQPSK_CHANNEL_MASK);
 
         length += sizeof(MeshCoP::ChannelMaskEntry);
-
-#if OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT
-        VerifyOrExit((entry = static_cast<MeshCoP::ChannelMaskEntry *>(entry->GetNext(this))) != NULL,
-                     error = OT_ERROR_NO_BUFS);
-#endif
```

Comment:
Why delete the code between 199 and 203? When OT supports both the 915MHz and 2.4GHz, the code between line 210 and 214 will cover the result of the code between 194 and 198. According to the definition of section 8.10.1.18 in Thread 1.1.1, the ChannelMask TLV should include a list of Channel Mask Entries when there are different channel pages.